operators/src/processing/reprojection.rs

```diff
@@ -160,16 +161,24 @@ where
     }
 }
 
-/// this method performs the reverse transformation of a query rectangle
+/// this method performs the transformation of a query rectangle in `target` projection
+/// to a new query rectangle with coordinates in the `source` projection
 pub fn query_rewrite_fn(
     query: QueryRectangle,
-    from: SpatialReference,
-    to: SpatialReference,
+    source: SpatialReference,
+    target: SpatialReference,
 ) -> Result<QueryRectangle> {
-    let projector = CoordinateProjector::from_known_srs(to, from)?;
-    let p_bbox = query.bbox.reproject(&projector)?;
-    let p_spatial_resolution =
-        suggest_pixel_size_from_diag_cross(query.bbox, query.spatial_resolution, &projector)?;
+    let projector_source_target = CoordinateProjector::from_known_srs(source, target)?;
+    let projector_target_source = CoordinateProjector::from_known_srs(target, source)?;
+
+    let p_bbox = query.bbox.reproject_clipped(&projector_target_source)?;
+    let s_bbox = p_bbox.reproject(&projector_source_target)?;
+
+    let p_spatial_resolution = suggest_pixel_size_from_diag_cross(
```

Comment:
you have both bounding boxes now. so we can avoid the projection in suggest_pixel_size_from_diag_cross... maybe add a non projecting version o

Reply:
ca3ae8b132458a61c13a109d75abd0d2569b3883